components/Modal/__tests__/__snapshots__/Modal.test.js.snap

```diff
@@ -2,66 +2,71 @@
 
 exports[`Modal should render with many props assigned 1`] = `
 <body
-  aria-hidden="true"
-  class="ReactModal__Body--open"
+  style="pointer-events: none;"
 >
-  <div
-    data-react-modal-body-trap=""
-    style="position: absolute; opacity: 0;"
+  <span
+    aria-hidden="true"
+    data-aria-hidden="true"
+    data-radix-focus-guard=""
+    style="outline: none; opacity: 0; position: fixed; pointer-events: none;"
     tabindex="0"
   />
-  <div />
   <div
-    data-react-modal-body-trap=""
-    style="position: absolute; opacity: 0;"
-    tabindex="0"
+    aria-hidden="true"
+    data-aria-hidden="true"
   />
   <div
-    class="Modal"
+    aria-hidden="true"
+    class="overlay"
+    data-aria-hidden="true"
+    data-state="open"
+    data-testid="MODAL_OVERLAY"
+    style="pointer-events: auto;"
   >
-    <div
-      class="ReactModal__Overlay"
-      style="position: fixed; top: 0px; left: 0px; right: 0px; bottom: 0px; background-color: rgba(255, 255, 255, 0.75);"
+    <button
+      class="CloseButton"
+      data-testid="CLOSE_BUTTON"
+      type="button"
```

Comment:
one thing that interests me is that Dialog.Close didn't actually affect the button, so somehow it's not working.

Reply:
hmmm what do you mean by that? I'm not sure what adding the wrapper does in terms of affecting any functionality.

Reply:
By doing

```tsx
<Dialog.Close asChild><Button {...props} /* close button */ /></Dialog.Close>
```

What happens is that props are spread onto <Button>, including:
- `aria-haspopup="dialog"`
- `aria-expanded="false"`
- `aria-controls="radix-30"`
- `data-state="closed"`

It's probably because we filter the props in the Button component, but I can fix that later 😁

Reply:
Ah that makes sense! Ty!